src/otx/algo/detection/heads/anchor_head.py

```diff
@@ -402,72 +391,15 @@ def get_targets(
-        labels: Tensor,
-        label_weights: Tensor,
-        bbox_targets: Tensor,
-        bbox_weights: Tensor,
-        avg_factor: int,
-    ) -> tuple:
-        """Calculate the loss of a single scale level based on the features extracted by the detection head.
-
-        TODO (kirill): it is related to RPNHead for iseg, will be deprecated
-
-        Args:
-            cls_score (Tensor): Box scores for each scale level
-                Has shape (N, num_anchors * num_classes, H, W).
-            bbox_pred (Tensor): Box energies / deltas for each scale
-                level with shape (N, num_anchors * 4, H, W).
-            anchors (Tensor): Box reference for each scale level with shape
-                (N, num_total_anchors, 4).
-            labels (Tensor): Labels of each anchors with shape
-                (N, num_total_anchors).
-            label_weights (Tensor): Label weights of each anchor with shape
-                (N, num_total_anchors)
-            bbox_targets (Tensor): BBox regression targets of each anchor
-                weight shape (N, num_total_anchors, 4).
-            bbox_weights (Tensor): BBox regression loss weights of each anchor
-                with shape (N, num_total_anchors, 4).
-            avg_factor (int): Average factor that is used to average the loss.
-
-        Returns:
-            tuple: loss components.
-        """
-        # classification loss
-        labels = labels.reshape(-1)
-        label_weights = label_weights.reshape(-1)
-        cls_score = cls_score.permute(0, 2, 3, 1).reshape(-1, self.cls_out_channels)
-        loss_cls = self.loss_cls(cls_score, labels, label_weights, avg_factor=avg_factor)  # type: ignore[misc] # TODO (kirill): fix
-        # regression loss
-        target_dim = bbox_targets.size(-1)
-        bbox_targets = bbox_targets.reshape(-1, target_dim)
-        bbox_weights = bbox_weights.reshape(-1, target_dim)
-        bbox_pred = bbox_pred.permute(0, 2, 3, 1).reshape(-1, self.bbox_coder.encode_size)
-        if self.reg_decoded_bbox:
-            # When the regression loss (e.g. `IouLoss`, `GIouLoss`)
-            # is applied directly on the decoded bounding boxes, it
-            # decodes the already encoded coordinates to absolute format.
-            anchors = anchors.reshape(-1, anchors.size(-1))
-            bbox_pred = self.bbox_coder.decode(anchors, bbox_pred)
-        loss_bbox = self.loss_bbox(bbox_pred, bbox_targets, bbox_weights, avg_factor=avg_factor)  # type: ignore[misc] # TODO (kirill): fix
-        return loss_cls, loss_bbox
-
-    def loss_by_feat(
+    def get_targets_for_loss(
         self,
         cls_scores: list[Tensor],
-        bbox_preds: list[Tensor],
         batch_gt_instances: list[InstanceData],
         batch_img_metas: list[dict],
         batch_gt_instances_ignore: list[InstanceData] | None = None,
-    ) -> dict:
+    ) -> tuple:
         """Calculate the loss based on the features extracted by the detection head.
```

Comment:
Could you update this docstring?

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/3865/commits/c3bc2458fb51a22f86f59b7ffc7a09d2acf0389f